plugin/storage/es/mappings/mapping_test.go

```diff
@@ -0,0 +1,316 @@
+// Copyright (c) 2021 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mappings
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"testing"
+	"text/template"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/jaegertracing/jaeger/pkg/es"
+	"github.com/jaegertracing/jaeger/pkg/es/mocks"
+)
+
+func TestMappingBuilder_GetMapping(t *testing.T) {
+	tests := []struct {
+		mapping   string
+		esVersion uint
+	}{
+		{mapping: "jaeger-span", esVersion: 7},
+		{mapping: "jaeger-span", esVersion: 6},
+		{mapping: "jaeger-service", esVersion: 7},
+		{mapping: "jaeger-service", esVersion: 6},
+		{mapping: "jaeger-dependencies", esVersion: 7},
+		{mapping: "jaeger-dependencies", esVersion: 6},
+	}
+	for _, tt := range tests {
+		t.Run(tt.mapping, func(t *testing.T) {
+			mb := &MappingBuilder{
+				TemplateBuilder: es.TextTemplateBuilder{},
+				Shards:          3,
+				Replicas:        3,
+				EsVersion:       tt.esVersion,
+				IndexPrefix:     "",
+				UseILM:          false,
+			}
+			got, err := mb.GetMapping(tt.mapping)
+			require.NoError(t, err)
+			want := ""
+			if tt.esVersion == 7 {
+				want, err = mb.fixMapping("/" + tt.mapping + "-7.json")
```

Comment:
Although these tests achieve the goal of line coverage, they appear execute the same logic between the `want` and `got`. This means that if there's a bug in, say, `fixMapping()`, we wouldn't catch it.

I think we ought to have a static file in a fixtures directory of `mappings` that represents the `want`ed resulting mapping, then assert against the contents of this file.

Reply:
Yep, agreed. Would add fixtures as suggested.

Reply:
Added in next commit